common/lib/dependabot/version.rb

```diff
@@ -1,21 +1,75 @@
-# typed: true
+# typed: strong
 # frozen_string_literal: true
 
+require "sorbet-runtime"
+
 module Dependabot
   class Version < Gem::Version
+    extend T::Sig
+    extend T::Helpers
+
+    abstract!
+
+    sig do
+      override
+        .overridable
+        .params(
+          version: T.any(
+            String,
+            Integer,
+            Float,
+            Gem::Version,
+            NilClass
+          )
+        )
+        .void
+    end
     def initialize(version)
-      @original_version = version
+      @original_version = T.let(version.to_s, String)
 
-      super
+      T.unsafe(super(version))
+    end
+
+    sig do
+      override
+        .overridable
+        .params(
+          version: T.any(
+            String,
+            Integer,
+            Float,
+            Gem::Version,
+            NilClass
+          )
+        )
+        .returns(Dependabot::Version)
+    end
+    def self.new(version)
+      T.cast(super, Dependabot::Version)
     end
```

Comment:
This override is required because we do `version_class.new`, and without the override, `super` calls `Gem::Version.new` which returns `Gem::Version` which doesn't have `self.correct?`.

Reply:
Weird but makes sense given that `Gem::Version` [overrides `::new`](https://github.com/rubygems/rubygems/blob/8ffa73df3a250623b13c5bf7ab48ed8b8e84e46f/lib/rubygems/version.rb#L203)